features/vaadin-dashlets/dashlet-ksc/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/KscDashlet.java

```diff
@@ -430,6 +425,19 @@ public Component getComponent() {
         });
     }
 
+    String determineNodeIdByResourceId(String resourceId){
+
+        // old implementation:
+        String[] arr0 = resourceId.split("\\.");
+        String[] arr1 = arr0[0].split("[\\[\\]]");
+        // data.put("nodeId", arr1[1]); // results in: Test:1525957453778 which is not a valid node id
+
+        // possible new solution but with ugly casting:
+        int onmsNodeId = ((OnmsNode)m_resourceDao.getResourceById(ResourceId.fromString(resourceId))
+                .getParent().getEntity()).getId();
```

Comment:
is there a better way to deduct the internal node id from a resourceId? The above code works but looks mighty ugly to me...

Reply:
There are some helper methods in ResourceTypeUtils and ResourceIdParser. Don't know If they can help at this point.

Reply:
+1 I'll take a look.